Build: osx_arm64

ax3l opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

ax3l committed:
Build: osx_arm64

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ax3l commented:
@conda-forge-admin, please rerender

ax3l commented on conda-forge.yml:
```diff
@@ -1,3 +1,10 @@
+build_platform:
+  linux_aarch64: default
+  linux_ppc64le: default
```
Only cross-compile `osx_arm64` for now:
```suggestion
```

ax3l commented:
@conda-forge-admin, please rerender

ax3l commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.7, conda-smithy 3.15.0, and conda-forge-pinning 2021.12.07.05.32.24

ax3l commented:
Cross-compile support is still lacking upstream: https://bitbucket.org/icl/blaspp/issues/13

ax3l committed:
Work-Around: Set BLAS (Cross)

Work-around https://bitbucket.org/icl/blaspp/issues/13 by setting the
BLAS libraries manually.

ax3l committed:
Modernize: `use_cuda` -> `gpu_backend`

```
CMake Deprecation Warning at CMakeLists.txt:73 (message):
  WARNING: `use_cuda=OFF` and `use_hip=` are deprecated; setting `gpu_backend
  = none`
```

ax3l committed:
Set: BLAS_LIBRARIES

Same as we do in Spack. Avoids all heuristics and avoids that, e.g.,
macOS picks up the accelerate BLAS bindings.

ax3l commented:
@conda-forge-admin, please rerender

ax3l committed:
Re-introduce work-around for cross
